File: baekjoon/baekjoon_10828.py
import sys
stack = []
N = int(sys.stdin.readline().rstrip())
for line in range(N):
    commands = sys.stdin.readline().rstrip().split()
    if(len(commands) > 1 and commands[0] == "push"):
        stack.append(int(commands[1]))
    else:
        command = commands[0]
        if command == "pop":
            if len(stack) == 0:
                print(-1)
            else:
                print(stack[-1])
                del stack[-1]
        elif command == "size":
            print(len(stack))
        elif command == "empty":
            if(len(stack) == 0):
                print(1)
            else:
                print(0)
        elif command == "top":
            if len(stack) == 0:
                print(-1)
            else:
                print(stack[-1])

File: baekjoon/baekjoon_10828.cpp
#include <bits/stdc++.h>
using namespace std;

int main(void) {
    ios::sync_with_stdio(0);
    cin.tie(0);
    int N;
    string op;
    int num;
    stack <int> S;
    cin >> N;
    for (int i=0; i<N; i++) {
        cin >> op;
        if (op == "push") {
            cin >> num;
            S.push(num);
        }
        else if (op == "pop") {
            if (S.size() == 0) {
                cout << -1 << "\n";
            }
            else {
                cout << S.top() << "\n";
                S.pop();
            }
        }
        else if (op == "size") {
            cout << S.size() << "\n";
        }
        else if (op == "empty") {
            if (S.size() == 0) {
                cout << 1 << "\n";
            }
            else {
                cout << 0 << "\n";
            }
        }
        else if (op == "top") {
            if (S.size() == 0) {
                cout << -1 << "\n";
            }
            else {
                cout << S.top() << "\n";
            }
        }
        
    }
}